crates/adkg-cli/src/main.rs

```diff
@@ -223,27 +210,131 @@ async fn run_adkg(args: RunAdkg) -> anyhow::Result<()> {
     };
 
     tracing::info!("Stopping libp2p dispatcher...");
-    dispatcher.stop().await;
+    transports.topic_dispatcher.stop().await;
 
     tracing::info!("Stopping libp2p transport...");
-    if let Err(e) = libp2p_node.stop().await {
+    if let Err(e) = transports.node.stop().await {
         tracing::error!(error = ?e, "Failed to stop libp2p node");
     }
 
-    match output {
-        Ok(output) => {
-            tracing::info!(used_sessions = ?output.used_sessions, "Successfully obtained secret key from ADKG");
-            write_adkg_keys(output, &priv_out, &pub_out, adkg_scheme_name, &group_config)?;
-        }
-        Err(e) => {
-            tracing::info!(error = ?e, "Failed to execute ADKG");
-            Err(e)?
-        }
+    process_adkg_output(
+        &priv_out,
+        &pub_out,
+        transcript_out,
+        &group_config,
+        adkg_scheme_name,
+        output,
+    )?;
+
+    Ok(())
+}
+
+async fn rescue_adkg(args: Rescue) -> anyhow::Result<()> {
+    let Rescue {
+        common:
+            AdkgRunCommon {
+                scheme,
+                group_file,
+                priv_file,
+                id,
+                priv_out,
+                pub_out,
+            },
+        transcript_files,
+    } = args;
+
+    // Parse common inputs
+    let (scheme_config, group_config, sk) =
+        parse_adkg_common(&scheme, &group_file, &priv_file, &priv_out, &pub_out)?;
+
+    // Parse transcripts
+    let transcripts = transcript_files
+        .into_iter()
+        .map(|transcript_file| {
+            fs::read(&transcript_file).with_context(|| {
+                format!("failed to read transcript `{}`", transcript_file.display())
+            })
+        })
+        .collect::<Result<Vec<_>, _>>()?;
+    if transcripts.len() < group_config.n.get() - group_config.t.get() {
+        Err(anyhow!(
+            "Not enough transcripts specified, expected number of transcripts >= n - t = {}",
```

Comment:
would suggest putting in the actual count here to help diagnose errors, e.g. "oh shit, it said 0 I must have pointed it to an empty file"